### ISSUE TITLE
Python: Distinguish constructor and constructor call

### ISSUE THREAD

**Issue by notEvil** [nvim-treesitter/nvim-treesitter#5482]
Hi,

currently `__init__` (and `__new__`) and upper case function calls match to `@constructor`. It would be nice to have them match to separate targets imo. What about `@constructor.call`?

**Comment by lucario387** [comment]
I don't see that alone as a valid argument for a specific capture.

**Comment by notEvil** [comment]
Its a matter of preference for sure. With `@function`/`@method` and `@function.call`/`@method.call` its an established theme and being able to distinguish "definition" from "call" is a good thing imo.

**Comment by lucario387** [comment]
Yes, but it's just python, and they are, well, constructors, most of the time you don't call them, you only declare what they are supposed to do. Java does call class constructors, but even jdtls or in intelliJ, they don't color the `Class` in `new Class` differently from `public Class`, so the need to differentiate definition from call is not really valid for it. This also applies to C++ and JS family. I'll change my mind if there's a more valid reasoning a different capture for this case needs to exist.

**Comment by notEvil** [comment]
People seem to accept the default experience and thats fine. In the meantime I realized that in order to tweak the highlights we have to patch the queries or replace them entirely (its not possible to add/replace/remove or reorder the patterns easily). So you can close this issue if you want.

**Comment by lucario387** [comment]
I wouldn't say it's "seem to have accept" or not. Just that I have yet to see of a more convincing opinion. Open to all suggestions.
